mne/time_frequency/tfr.py

```diff
@@ -459,127 +682,259 @@ def single_trial_power(data, sfreq, frequencies, use_fft=True, n_cycles=7,
+                       output=output, decim=decim, **tfr_params)
+    times = inst.times[decim].copy()
+
+    if average:
+        if return_itc:
+            power, itc = out.real, out.imag
+        else:
+            power = out
+        nave = len(data)
+        out = AverageTFR(info, power, times, freqs, nave,
+                         method='%s-power' % method)
+        if return_itc:
+            out = (out, AverageTFR(info, itc, times, freqs, nave,
+                                   method='%s-itc' % method))
+    else:
+        power = out
+        out = EpochsTFR(info, power, times, freqs, method='%s-power' % method)
+
+    return out
+
 
-    The time frequency decomposition is done with Morlet wavelets
+@verbose
+def tfr_morlet(inst, freqs, n_cycles, use_fft=False, return_itc=True, decim=1,
+               n_jobs=1, picks=None, zero_mean=True, average=True,
+               verbose=None):
+    """Compute Time-Frequency Representation (TFR) using Morlet wavelets
 
     Parameters
     ----------
-    data : array
-        3D array of shape [n_epochs, n_channels, n_times]
-    sfreq : float
-        Sampling frequency.
-    frequencies : array
-        Array of frequencies of interest
-    use_fft : bool
-        Compute transform with fft based convolutions or temporal
-        convolutions.
-    n_cycles : float | array of float
-        Number of cycles. Fixed number or one per frequency.
-    decim : int | slice
+    inst : Epochs | Evoked
+        The epochs or evoked object.
+    freqs : ndarray, shape (n_freqs,)
+        The frequencies in Hz.
+    n_cycles : float | ndarray, shape (n_freqs,)
+        The number of cycles globally or for each frequency.
+    use_fft : bool, defaults to False
+        The fft based convolution or not.
+    return_itc : bool, defaults to True
+        Return inter-trial coherence (ITC) as well as averaged power.
+        Must be ``False`` for evoked data.
+    decim : int | slice, defaults to 1
         To reduce memory usage, decimation factor after time-frequency
         decomposition.
         If `int`, returns tfr[..., ::decim].
         If `slice` returns tfr[..., decim].
-        Note that decimation may create aliasing artifacts.
-        Defaults to 1.
-    n_jobs : int
-        The number of CPUs used in parallel. All CPUs are used in -1.
-        Requires joblib package.
-    zero_mean : bool
-        Make sure the wavelets are zero mean.
+        .. note: Decimation may create aliasing artifacts.
+    n_jobs : int, defaults to 1
+        The number of jobs to run in parallel.
+    picks : array-like of int | None, defaults to None
+        The indices of the channels to plot. If None, all available
+        channels are displayed.
+    zero_mean : bool, defaults to True
+        Make sure the wavelet has a mean of zero.
+
+        .. versionadded:: 0.12.0
+    average : bool, defaults to True
+        If True average accross Epochs.
+
+        .. versionadded:: 0.12.0
```

Comment:
ahh, because it used to be `average=True` effectively. I say we deprecate the param to get people to do `fun(...).average()` instead.